test/zdtm/lib/ns.c

```diff
@@ -207,6 +209,36 @@ static void ns_sig_hand(int signo)
 	write(STDERR_FILENO, buf, MIN(len, sizeof(buf)));
 }
 
+#ifndef CLONE_NEWTIME
+#define CLONE_NEWTIME   0x00000080      /* New time namespace */
+#endif
+
+static inline int _settime(clockid_t clk_id, time_t offset)
+{
+        int fd, len;
+        char buf[4096];
+
+        if (clk_id == CLOCK_MONOTONIC_COARSE || clk_id == CLOCK_MONOTONIC_RAW)
+                clk_id = CLOCK_MONOTONIC;
+
+        len = snprintf(buf, sizeof(buf), "%d %ld 0", clk_id, offset);
+
+        fd = open("/proc/self/timens_offsets", O_WRONLY);
+        if (fd < 0) {
+                fprintf(stderr, "/proc/self/timens_offsets: %m");
+		return -1;
+	}
+
+        if (write(fd, buf, len) != len) {
+                fprintf(stderr, "/proc/self/timens_offsets: %m");
+		return -1;
+	}
+
+        close(fd);
+
+        return 0;
+}
```

Comment:
nit: something with the indention here: are those spaces?
also, fprintf() stands out